services/graphql-server/utils/ast2sdl.go

```diff
@@ -2,72 +2,119 @@ package utils
 
 import (
 	"fmt"
+	"reflect"
 	"strings"
 
 	"github.com/graphql-go/graphql"
 	"github.com/graphql-go/graphql/language/ast"
+	"github.com/sirupsen/logrus"
 )
 
-// ResolveParamsToSDLQuery - generates SDL query from graphql.ResolveParams object
-func ResolveParamsToSDLQuery(queryName string, rp graphql.ResolveParams, args string) string {
-	builder := strings.Builder{}
+// GqlRequestConfig - type to store all data for gql request
+type GqlRequestConfig struct {
+	Query          string
+	VariableValues map[string]interface{}
+}
+
+type sdlQuery struct {
```

Comment:
Unclear on two very similarly named fields - `fragmentsToAdd` and `addedFragments`

Reply:
One fragment can be dependent on another one and several fragments can be dependent from one. So I need to manage what fragments I need to add to query and what fragment already was added. So `fragmentToAdd` stores fragments that should be added to query and `addedFragments` contains fragments that already were added to query string

Reply:
So we track all fragments we run into while traversing the AST, and when we finish we append the fragments clauses?
Sounds like we should only need one non-duplicate list of all fragments we run into for that, no? If we append them all at once at the end, why do we need to track which ones we've already added?

Reply:
I'll try to explain by example.
1) Query refers to fragments A and B
2) Fragment A refers no other fragments
3) Fragment B refers to fragment A and C.
4) Fragment C reference no fragments. 

After building query we have A and B in list.
Then we add A and B fragment clauses to the whole string.
During addition of B we see that there is reference to A and C.
But A fragment was already added to the result string and C wasn't
So in addition to the set (a.k.a. non-duplicate list) we need something that can tell us if the fragment was already added or need to be added.

Reply:
Talked offline, we settled on creating a struct that will abstract the map-array weirdness with some methods like "SeenFragment" "AddedFragment" etc